### ISSUE TITLE
Links not clickable

### ISSUE THREAD

**Issue by d-sko** [nextcloud/notes#173]
Hello, I've installed this app in my nextcloud and noticed a problem: I have some notes with links in it and they are not clickable (but they are actually displayed as links). Is this intended or am I doing something wrong?

![nc_notes_link](https://user-images.githubusercontent.com/5774313/36891566-63342ee4-1e02-11e8-846d-c5bca9292489.PNG)

**Comment by d-sko** [comment]
Oh, sorry I missed that.

Closing because duplicate of #158

**Comment by jancborchardt** [comment]
Reopening cause this issue is specifically about that one problem. :)

**Comment by stefan-niedermann** [comment]
Have you tried `ctrl` + click? I thought this would work...

**Comment by tribut** [comment]
`ctrl`+ click works, but it also creates a second cursor (a feature which I was not aware the markdown editor had).

**Comment by tribut** [comment]
Note that `ctrl` + click only works on manually set up link (`[name](target)`) while it seems to do nothing except creating multiple cursors on auto-detected urls.

**Comment by stefan-niedermann** [comment]
i see. single click is disabled like in microsoft word and libreoffice writer, because if one wants to edit the link or the text, one could only open the link and not set the cursor.

**Comment by jancborchardt** [comment]
@stefan-niedermann I’d still say we should enable single-click because Ctrl-Click is not really discoverable. If the link should be edited, people can still click next to it and use the arrow keys – I know, also not ideal, but less bad than not being able to click the link :\ Is that possible?

**Comment by stefan-niedermann** [comment]
Technically yes. But i would recommend to wait for the text-pop-over: Click on a link could create a pop over with an "open"-button. this way one could still click into links to edit them. What do you think?

Every other text-editor that is capable of displaying and opening links uses `ctrl` + click to open the links...

**Comment by jancborchardt** [comment]
> Every other text-editor that is capable of displaying and opening links uses ctrl + click to open the links...

That's only the case for mostly code editors. :) FoldingText and iA Writer for example open links on click, and I do agree it's confusing from a user perspective that click doesn't open a link.

The popover from #200 is also only intended to come when you mark text, not when you click.

**Comment by tribut** [comment]
Actually, LibreOffice only displays a tooltip on hover:

![libreoffice-writer](https://user-images.githubusercontent.com/719105/41034573-8060de0a-6979-11e8-8b78-175fb6791de7.png)

Same behavior on Word (2010):

![word-2010](https://user-images.githubusercontent.com/719105/41034689-eaf8f004-6979-11e8-8dad-e3ca5e466252.png)

If "Rightclick -> Open in new tab/window" works, I would argue ctrl+click is actually fine and consistent with other editors.

**Comment by mmccarn** [comment]
I would add to this that on OSX (High Sierra), <ctrl>-click is intercepted by the OS and interpreted as a right-click, which gives me the browser right-click menu.  

<Command⌘>-click does insert extra cursors as described by tribut but still doesn't open the link.

**Comment by jancborchardt** [comment]
Again:

> FoldingText and iA Writer for example open links on click, and I do agree it's confusing from a user perspective that click doesn't open a link.


I would say that neither LibreOffice nor Word are a really good benchmark when it comes to distraction-free working.


Right now we have the problem that you can not click the link _at all_ and it is very cumbersome to get to it. Ctrl-click is not possible on touch-interfaces, or in touch-modes of desktop computers (like Surface etc). Hence the easiest way to fix this for now is to make clicking open the link. As you can then still modify the link by clicking next to it and then moving your cursor, or by marking, or if possible we could use ctrl-click for that.

The point being that after insertion, you probably _click_ a link more often than you edit it. And yes, in Markdown-format links like `[Text](https://…)` clicking the text part of the link should _not_ open the link so it’s easy to edit the text.

**Comment by brandonkirsch** [comment]
Weighing in on an old issue.

On my web browser:  Ctrl+Click only follows a hyperlink when I click on one that has been formatted in Markdown properly.

I find it extremely annoying that I cannot Ctrl+Click a hyperlink that was formatted automatically when using my Desktop web browser to access Notes.

Anyone else?

**Comment by monperrus** [comment]
Yes, it would be great that URLs are automatically recognized as such and made clickable (as tomboy and many other note taking apps do).
